lib/hammer_cli_foreman/commands.rb

```diff
@@ -279,6 +279,86 @@ def should_retrieve_all?
 
   end
 
+  class ListSearchCommand < ListCommand
+    def self.search_resource(res, action = :index)
+      resource res, action
+      default_search_options
+    end
+
+    def self.default_search_options
+      option("--#{module_resource.singular_name}-id".tr('_', '-'), "ID", _("%s Id") % module_resource.singular_name)
```

Comment:
I tried to convert `host reports` into descendant of this class. The resulting command looks good and easy to read. However in hammer we traditionally use `--id` and `--name` to refer to the "module resource" in this type of commands i.e. `host reports --id` and `host reports --name`. Would you mind to change the option names accordingly? It may make sense to add a mapping to change it if you have a use case for that.